GitCommands/Git/GitRevision.cs

```diff
@@ -86,6 +82,77 @@ public BuildInfo BuildStatus
         public string Subject { get; set; } = "";
         [CanBeNull]
         public string Body { get; set; }
+
+        [CanBeNull]
+        public string BodySummary
+        {
+            get
+            {
+                if (Body == null || !HasMultiLineMessage)
+                {
+                    return Body;
+                }
+
+                return _bodySummary ?? (_bodySummary = BuildSummary(Body.TrimEnd()));
+            }
+        }
+
+        private const int MaxLineLength = 150;
+        private static string BuildSummary(string body)
+        {
+            var s = new StringBuilder(100);
```

Comment:
Do I understand correctly we expect to render max ~4750-4800 chars?
```cs
int worstCaseLength = (/*max line length*/150 + /* [...]*/6 + /*CR*/1) * 30 + /*last [...]CR*/7 + /*CR*/1;
```

Should we then allocate like this?
```suggestion
            var s = new StringBuilder(Math.Min(body.Length, worstCaseLength) + /*branch name*/???);
```

Reply:
This is still unanswered

Reply:
If we want to optimize that, I'm also tempted to create only static instance of `StringBuilder` and surround the usage with a `lock() {...}`.

What do you think?

Reply:
Because thinking about it again, allocating the worst case length is **very** pessimistic even when the body is more long than this limit.

Reply:
> I'm also tempted to create only static instance of `StringBuilder` and surround the usage with a `lock() {...}`.

`GitRevision` isn't a singleton, we have multiple of those floating around. Introducing a singleton here will lead to race conditions and a may become a bottleneck.

Reply:
done.